src/state/feed-feedback.tsx

```diff
@@ -0,0 +1,145 @@
+import React from 'react'
+import {AppState, AppStateStatus} from 'react-native'
+import {AppBskyFeedDefs, BskyAgent} from '@atproto/api'
+import debounce from 'lodash.debounce'
+
+import {PROD_DEFAULT_FEED} from '#/lib/constants'
+import {logger} from '#/logger'
+import {FeedDescriptor, isFeedPostSlice} from '#/state/queries/post-feed'
+import {useAgent} from './session'
+
+type StateContext = {
+  enabled: boolean
+  onItemSeen: (item: any) => void
+  sendInteraction: (interaction: AppBskyFeedDefs.Interaction) => void
+  flushAndReset: () => void
+}
+
+const stateContext = React.createContext<StateContext>({
+  enabled: false,
+  onItemSeen: (_item: any) => {},
+  sendInteraction: (_interaction: AppBskyFeedDefs.Interaction) => {},
+  flushAndReset: () => {},
+})
+
+export function useFeedFeedback(feed: FeedDescriptor, hasSession: boolean) {
+  const {getAgent} = useAgent()
+  const enabled = isDiscoverFeed(feed) && hasSession
+  const queue = React.useRef<Set<string>>(new Set())
+  const history = React.useRef<Set<string>>(new Set())
+
+  const [sendToFeed] = React.useState(() =>
```

Comment:
This is a subtle pitfall because if you use any state or props inside this function (or if you call any method that uses those), they'll be stale. Since the initial state isn't reactive.

I think I'll probably restructure this, need to think how.